pkg/render/tiers/tiers.go

```diff
@@ -69,7 +74,15 @@ func (t tiersComponent) Objects() ([]client.Object, []client.Object) {
 		t.allowTigeraClusterDNSPolicy(),
 	}
 
-	return objsToCreate, nil
+	objsToDelete := []client.Object{}
+
+	if t.cfg.DNSEgressCIDRs != nil && len(t.cfg.DNSEgressCIDRs) > 0 {
```

Comment:
In Golang, for length checks you don't actually need to do a `nil` check first - `len` will return 0 if the slice is nil, so you can simplify this if statement.

https://go.dev/play/p/6bAfsebsrLB